packages/react-dom/src/events/ReactDOMEventListener.js

```diff
@@ -358,10 +330,10 @@ function dispatchEventWithEnableCapturePhaseSelectiveHydrationWithoutDiscreteEve
       }
       blockedOn = nextBlockedOn;
     }
-    if (blockedOn) {
+    if (blockedOn !== null) {
       nativeEvent.stopPropagation();
-      return;
     }
+    return;
```

Comment:
I'm worried that removing this return doesn't break any tests. I think it's correct to have it but we should follow-up to have coverage against over-dispatching.

Reply:
I think the real problem is that removing  lines 341 - 347:
```
dispatchEventForPluginEventSystem(
    domEventName,
    eventSystemFlags,
    nativeEvent,
    null,
    targetContainer,
  );
 ```
 
 Didn't cause any tests to fail. Honestly, tracing through the code, dispatching with a `null` target does not seem to accomplish anything since it would lead to 0 listeners being executed O.o